### PR TITLE
AMPI Benchmarks: Cleanup and add untested tests

### PR DESCRIPTION
**Pull request by nitbhat**
The major change in this PR is the addition of the following tests to the `test` target:
alltoall  isomalloc onesided speed

The minor change involves cleanup of the trailing white spaces and indentation from source code files. I also ran clang-format on the source code files.

### CODE REVIEW

File: benchmarks/ampi/alltoall/Makefile [hunk 1]
```diff
@@ -23,11 +24,11 @@ alltoall.mpi: alltoall.c
```
Comment (nitbhat) [inline benchmarks/ampi/alltoall/Makefile:24]: Should we run these tests? (And also alltoall_VPtest).

**Comment by nitbhat** [comment]
It looks like we should change our `.clang-format` to incorporate those stylistic changes that you made for `isomalloc.C`

**Comment by evan-charmworks** [comment]
I fixed the CUDA buildold failure. We should not build MPI programs as part of `make` or `make test`. They are still useful to have in the Makefile in order to compare with conventional MPI implementations.

> I also see some CMA related failures when I run the tests on my lab machine. 

This sounds like a bug, probably never exposed because none of the development machines I have used and none of our automated testing has CMA enabled.

> It looks like we should change our `.clang-format` to incorporate those stylistic changes that you made for `isomalloc.C`. @evan-charmworks

That is an uphill battle I don't feel like fighting. In 164cd19a17d296a91723135c39e1737335e81671 I got the most important change merged and discussion about `.clang-format` since then has been "Why did we change it from Attach? We use that all over the codebase. We should change it back." I take care in writing readable code and as long as it doesn't get mangled by clang-format's one-size-fits-all approach plus the suboptimal settings inherited from Google's specification, I won't rock the boat.

File: benchmarks/ampi/alltoall/Makefile (hunk 1 above)
Comment (evan-charmworks) [reply benchmarks/ampi/alltoall/Makefile:25]: As long as they don't take an excessive amount of time to complete, I think that would be good.

**Comment by evan-charmworks** [comment]
Build failure unrelated to PR:

```
make[5]: Entering directory '/home/circleci/project/netlrts-linux-x86_64/examples/charm++/zerocopy/entry_method_post_api/prereg/simpleZeroCopy'
../../../../../../bin/charmc  -c simpleZeroCopy.ci
touch cifiles
../../../../../../bin/charmc  -c simpleZeroCopy.C
../../../../../../bin/charmc  simpleZeroCopy.o -o simpleZeroCopy -language charm++ -module CommonLBs
../../../../../../bin/testrun  +p4 ./simpleZeroCopy 32 +balancer RotateLB ++local
Charmrun> scalable start enabled. 
Charmrun> started all node programs in 0.007 seconds.
Charm++> Running in non-SMP mode: 4 processes (PEs)
Converse/Charm++ Commit ID: v6.11.0-devel-302-g2b706a5f8
Isomalloc> Synchronized global address space.
Charm++> scheduler running in netpoll mode.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (1 sockets x 1 cores x 2 PUs = 2-way SMP)
Charm++> cpu topology info is gathered in 0.046 seconds.
[0] TreeLB in LEGACY MODE support
[0] TreeLB: Using PE_Root tree with strategy Rotate
send: completed
------------- Processor 1 Exiting: Caught Signal ------------
Reason: segmentation violation
Suggestion: Try running with '++debug', or linking with '-memory paranoid' (memory paranoid requires '+netpoll' at runtime).
[1] Stack Traceback:
  [1:0] libc.so.6 0x7ffff6d834b0 
  [1:1] libc.so.6 0x7ffff6e9c0e6 
  [1:2] simpleZeroCopy 0x4c7dea CkIndex_zerocopyObject::_call_zerocopySend_marshall6(void*, void*)
  [1:3] simpleZeroCopy 0x5485f4 CkDeliverMessageFree
zerocopySend: completed
  [1:4] simpleZeroCopy 0x58771d CkLocRec::invokeEntry(CkMigratable*, void*, int, bool)
  [1:5] simpleZeroCopy 0x54bd91 _processHandler(void*, CkCoreState*)
  [1:6] simpleZeroCopy 0x5fca93 CsdScheduleForever
  [1:7] simpleZeroCopy 0x5fd08d CsdScheduler
  [1:8] simpleZeroCopy 0x5f789a ConverseInit
  [1:9] simpleZeroCopy 0x5d2287 charm_main
mixedSend: completed
  [1:10] libc.so.6 0x7ffff6d6e830 __libc_start_main
  [1:11] simpleZeroCopy 0x4c3ab9 _start
Fatal error on PE 1> segmentation violation

real	0m0.137s
user	0m0.002s
sys	0m0.003s
Makefile:18: recipe for target 'test' failed
make[5]: *** [test] Error 1
make[5]: Leaving directory '/home/circleci/project/netlrts-linux-x86_64/examples/charm++/zerocopy/entry_method_post_api/prereg/simpleZeroCopy'
Makefile:19: recipe for target 'test-simpleZeroCopy' failed
```

**Comment by nitbhat** [comment]
> Build failure unrelated to PR:
> 
> ```
> make[5]: Entering directory '/home/circleci/project/netlrts-linux-x86_64/examples/charm++/zerocopy/entry_method_post_api/prereg/simpleZeroCopy'
> ../../../../../../bin/charmc  -c simpleZeroCopy.ci
> touch cifiles
> ../../../../../../bin/charmc  -c simpleZeroCopy.C
> ../../../../../../bin/charmc  simpleZeroCopy.o -o simpleZeroCopy -language charm++ -module CommonLBs
> ../../../../../../bin/testrun  +p4 ./simpleZeroCopy 32 +balancer RotateLB ++local
> Charmrun> scalable start enabled. 
> Charmrun> started all node programs in 0.007 seconds.
> Charm++> Running in non-SMP mode: 4 processes (PEs)
> Converse/Charm++ Commit ID: v6.11.0-devel-302-g2b706a5f8
> Isomalloc> Synchronized global address space.
> Charm++> scheduler running in netpoll mode.
> CharmLB> Load balancer assumes all CPUs are same.
> Charm++> Running on 1 hosts (1 sockets x 1 cores x 2 PUs = 2-way SMP)
> Charm++> cpu topology info is gathered in 0.046 seconds.
> [0] TreeLB in LEGACY MODE support
> [0] TreeLB: Using PE_Root tree with strategy Rotate
> send: completed
> ------------- Processor 1 Exiting: Caught Signal ------------
> Reason: segmentation violation
> Suggestion: Try running with '++debug', or linking with '-memory paranoid' (memory paranoid requires '+netpoll' at runtime).
> [1] Stack Traceback:
>   [1:0] libc.so.6 0x7ffff6d834b0 
>   [1:1] libc.so.6 0x7ffff6e9c0e6 
>   [1:2] simpleZeroCopy 0x4c7dea CkIndex_zerocopyObject::_call_zerocopySend_marshall6(void*, void*)
>   [1:3] simpleZeroCopy 0x5485f4 CkDeliverMessageFree
> zerocopySend: completed
>   [1:4] simpleZeroCopy 0x58771d CkLocRec::invokeEntry(CkMigratable*, void*, int, bool)
>   [1:5] simpleZeroCopy 0x54bd91 _processHandler(void*, CkCoreState*)
>   [1:6] simpleZeroCopy 0x5fca93 CsdScheduleForever
>   [1:7] simpleZeroCopy 0x5fd08d CsdScheduler
>   [1:8] simpleZeroCopy 0x5f789a ConverseInit
>   [1:9] simpleZeroCopy 0x5d2287 charm_main
> mixedSend: completed
>   [1:10] libc.so.6 0x7ffff6d6e830 __libc_start_main
>   [1:11] simpleZeroCopy 0x4c3ab9 _start
> Fatal error on PE 1> segmentation violation
> 
> real	0m0.137s
> user	0m0.002s
> sys	0m0.003s
> Makefile:18: recipe for target 'test' failed
> make[5]: *** [test] Error 1
> make[5]: Leaving directory '/home/circleci/project/netlrts-linux-x86_64/examples/charm++/zerocopy/entry_method_post_api/prereg/simpleZeroCopy'
> Makefile:19: recipe for target 'test-simpleZeroCopy' failed
> ```


Looks like this failure is the same as https://github.com/UIUC-PPL/charm/issues/2763. I'll try fixing that by next week.

**Comment by nitbhat** [comment]
> I fixed the CUDA buildold failure. We should not build MPI programs as part of `make` or `make test`. They are still useful to have in the Makefile in order to compare with conventional MPI implementations.
> 

Yes, makes sense. Thanks for fixing that. 


> This sounds like a bug, probably never exposed because none of the development machines I have used and none of our automated testing has CMA enabled.

Okay, I've opened this issue (https://github.com/UIUC-PPL/charm/issues/2962) and targetted it to 6.11. 

> > It looks like we should change our `.clang-format` to incorporate those stylistic changes that you made for `isomalloc.C`. @evan-charmworks
> 
> That is an uphill battle I don't feel like fighting. In [164cd19](https://github.com/UIUC-PPL/charm/commit/164cd19a17d296a91723135c39e1737335e81671) I got the most important change merged and discussion about `.clang-format` since then has been "Why did we change it from Attach? We use that all over the codebase. We should change it back." I take care in writing readable code and as long as it doesn't get mangled by clang-format's one-size-fits-all approach plus the suboptimal settings inherited from Google's specification, I won't rock the boat.

Yes, I see your point. However, I think formatting is more applicable for unformatted old code that definitely requires some sort of formatting and a decent `.clang-format` can help that cause in my opinion. Maybe worth discussing this in the next core meeting.

File: benchmarks/ampi/alltoall/Makefile [hunk 2]
```diff
@@ -37,5 +39,11 @@ alltoall_VPtest_f: alltoall_VPtest.f
 	$(CHARMF) -c alltoall_VPtest.f -swapglobals  $(OPTS)
 	$(CHARMF) -o alltoall_VPtest_f alltoall_VPtest.o -swapglobals $(OPTS) $(LIBS)
 
+test: all
+	$(call run, +p4 ./mpibench +vp2)
```
Comment (evan-charmworks) [inline benchmarks/ampi/alltoall/Makefile:43]: Should this be `+vp4`?

Comment (nitbhat) [reply benchmarks/ampi/alltoall/Makefile]: Yes, that's right. Fixed in the latest commit. Also added another run with +vp8.

File: benchmarks/ampi/alltoall/Makefile (hunk 1 above)
Comment (nitbhat) [reply benchmarks/ampi/alltoall/Makefile:24]: `alltoall` and `allgather` test are crashing on my machine, so I haven't added them yet. I've opened this issue for that (https://github.com/UIUC-PPL/charm/issues/2974). 

I added `alltoall_VPtest` to both test and testp in the latest commit.

File: benchmarks/ampi/alltoall/Makefile [hunk 3]
```diff
@@ -1,9 +1,10 @@
+-include ../../common.mk
 OPTS = -O3
 CHARMC = ../../../bin/ampicc
-CHARMF = ../../../bin/ampif90
 MPICC = mpicc
 
-all: mpibench mpibench.mpi
+all: mpibench allgather alltoall alltoall_VPtest
+all-mpi: mpibench.mpi allgather.mpi alltoall.mpi
```
Comment (evan-charmworks) [inline benchmarks/ampi/alltoall/Makefile:7]: Do `allgather.mpi alltoall.mpi` exist?

Comment (nitbhat) [reply benchmarks/ampi/alltoall/Makefile:7]: Didn't entirely get you. 

They are valid targets in the Makefile. If you're asking if they run successfully, they do. 
```

nbhat4@courage:/scratch/nitin/charm/mpi-linux-x86_64-debug/benchmarks/ampi/alltoall$ make test-mpi
mpirun -n 4 ./mpibench.mpi
#Alltoall: P=4, NUMTIMES=1, REPS=2000
4 2.7415 2.84 3.0295 2.805 3.761 15.8545 36.5685 69.284 880.196
mpirun -n 4 ./allgather.mpi 1024
Starting benchmark on 4 processors with 100 iterations
  100    1024	    0.004 msec,	 11377.396 Mbits/sec
mpirun -n 4 ./alltoall.mpi 1024
Starting benchmark on 4 processors with 1000 iterations
 1000    1024	    0.004 msec,	 4625.627 Mbits/sec
```

Comment (evan-charmworks) [reply benchmarks/ampi/alltoall/Makefile:7]: Ah, I missed the definition of their build targets. Should they be added to the `clean` target too?

File: benchmarks/ampi/alltoall/Makefile [hunk 4]
```diff
@@ -23,19 +24,38 @@ alltoall.mpi: alltoall.c
...
+	$(call run, +p$(P) ./mpibench +vp$(P) )
+	$(call run, +p$(P) ./mpibench +vp$$(( $(P) * 2 )) )
+	$(call run, +p$(P) ./mpibench +vp$$(( $(P) * 4 )) )
+	$(call run, +p$(P) ./alltoall_VPtest 1024 +vp$(P) )
+	$(call run, +p$(P) ./alltoall_VPtest 1024 +vp$$(( $(P) * 2 )) )
+	$(call run, +p$(P) ./alltoall_VPtest 1024 +vp$$(( $(P) * 4 )) )
 
 clean:
-	rm -rf *~ *.o alltoall mpibench charmrun conv-host allgather ampirun
+	rm -rf *~ *.o alltoall mpibench mpibench.mpi charmrun conv-host allgather ampirun
```
Comment (evan-charmworks) [inline benchmarks/ampi/alltoall/Makefile:61]:
```suggestion
	rm -rf *~ *.o alltoall alltoall.mpi mpibench mpibench.mpi charmrun conv-host allgather allgather.mpi ampirun
```

**Comment by evan-charmworks** [comment]
multicore darwin tracing failure:

```
Running command: ./alltoall_VPtest  +p4 1024 +vp8

Charm++: standalone mode (not using charmrun)
Charm++> Running in Multicore mode: 4 threads (PEs)
Converse/Charm++ Commit ID: 0626c52
Charm++: Tracemode Projections enabled.
Trace: traceroot: ./alltoall_VPtest
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (2 sockets x 2 cores x 1 PUs = 4-way SMP)
Charm++> cpu topology info is gathered in 0.000 seconds.
../../../bin/testrun: line 62: 47169 Abort trap: 6           ./charmrun "$@"

real	0m0.115s
user	0m0.178s
sys	0m0.228s
make[2]: *** [test] Error 134
make[1]: *** [test-alltoall] Error 2
make: *** [test-ampi] Error 2
##[error]Process completed with exit code 2.
```

**Comment by evan-charmworks** [comment]
Got a similar error with an MPI build:

```
../../../bin/testrun  +p4 ./alltoall_VPtest 1024 +vp8 

Running on 4 processors:  ./alltoall_VPtest 1024 +vp8 
charmrun>  /usr/bin/setarch x86_64 -R  mpirun -np 4  ./alltoall_VPtest 1024 +vp8 
Charm++> Running on MPI version: 3.1
Charm++> level of thread support used: -1 (desired: 0)
Charm++> Running in non-SMP mode: 4 processes (PEs)
Converse/Charm++ Commit ID: v6.11.0-devel-333-g6e7d83664
Isomalloc> Synchronized global address space.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 hosts (1 sockets x 4 cores x 2 PUs = 8-way SMP)
Charm++> cpu topology info is gathered in 0.000 seconds.
------------- Processor 0 Exiting: Called CmiAbort ------------
Reason: [0] Assertion "backend.isInRange(ptr)" failed in file /home/evan/Charmworks/charm/src/conv-core/isomalloc.C line 1131.

[0] Stack Traceback:
  [0:0] alltoall_VPtest 0x555555a103e6 CmiAbortHelper(char const*, char const*, char const*, int, int)
  [0:1] alltoall_VPtest 0x555555a104c0 CmiGetNonLocal
  [0:2] alltoall_VPtest 0x555555a1b646 CmiCopyMsg
  [0:3] alltoall_VPtest 0x555555a2ef87 isomalloc_dlmalloc::free(void*)
  [0:4] alltoall_VPtest 0x555555a269b0 CmiIsomallocContextFree
  [0:5] alltoall_VPtest 0x555555a0a8bd 
  [0:6] alltoall_VPtest 0x555555a0aca9 free
  [0:7] alltoall_VPtest 0x555555878ad2 __gnu_cxx::new_allocator<std::__detail::_Hash_node_base*>::deallocate(std::__detail::_Hash_node_base**, unsigned long)
  [0:8] alltoall_VPtest 0x55555587624c std::allocator_traits<std::allocator<std::__detail::_Hash_node_base*> >::deallocate(std::allocator<std::__detail::_Hash_node_base*>&, std::__detail::_Hash_node_base**, unsigned long)
  [0:9] alltoall_VPtest 0x55555599a492 std::__detail::_Hashtable_alloc<std::allocator<std::__detail::_Hash_node<std::pair<unsigned long const, int>, false> > >::_M_deallocate_buckets(std::__detail::_Hash_node_base**, unsigned long)
  [0:10] alltoall_VPtest 0x5555559963b4 std::_Hashtable<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_deallocate_buckets(std::__detail::_Hash_node_base**, unsigned long)
  [0:11] alltoall_VPtest 0x555555993242 std::_Hashtable<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_deallocate_buckets()
  [0:12] alltoall_VPtest 0x55555599f3cd std::_Hashtable<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_rehash_aux(unsigned long, std::integral_constant<bool, true>)
  [0:13] alltoall_VPtest 0x55555599bcf9 std::_Hashtable<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_rehash(unsigned long, unsigned long const&)
  [0:14] alltoall_VPtest 0x555555997937 std::_Hashtable<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_insert_unique_node(unsigned long, unsigned long, std::__detail::_Hash_node<std::pair<unsigned long const, int>, false>*)
  [0:15] alltoall_VPtest 0x55555599405b std::__detail::_Map_base<unsigned long, std::pair<unsigned long const, int>, std::allocator<std::pair<unsigned long const, int> >, std::__detail::_Select1st, std::equal_to<unsigned long>, std::hash<unsigned long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true>, true>::operator[](unsigned long const&)
  [0:16] alltoall_VPtest 0x555555991c03 std::unordered_map<unsigned long, int, std::hash<unsigned long>, std::equal_to<unsigned long>, std::allocator<std::pair<unsigned long const, int> > >::operator[](unsigned long const&)
  [0:17] alltoall_VPtest 0x55555597db04 CkLocMgr::deliverUnknown(CkArrayMessage*, CkArrayIndex const*, CkDeliver_t, int)
  [0:18] alltoall_VPtest 0x55555597d3e7 CkLocMgr::deliverMsg(CkArrayMessage*, CkArrayID, unsigned long, CkArrayIndex const*, CkDeliver_t, int)
  [0:19] alltoall_VPtest 0x55555597d760 CkLocMgr::sendMsg(CkArrayMessage*, CkArrayID, CkArrayIndex const&, CkDeliver_t, int)
  [0:20] alltoall_VPtest 0x555555958b7e CkArray::deliver(CkMessage*, CkArrayIndex const&, CkDeliver_t, int)
  [0:21] alltoall_VPtest 0x555555949e78 CProxyElement_ArrayBase::ckSend(CkArrayMessage*, int, int) const
  [0:22] alltoall_VPtest 0x55555580ec90 CProxyElement_ArrayElement::ckSend(CkArrayMessage*, int, int) const
  [0:23] alltoall_VPtest 0x5555558579ae CProxyElement_ampi::ckSend(CkArrayMessage*, int, int) const
  [0:24] alltoall_VPtest 0x555555841d82 CProxyElement_ampi::generic(AmpiMsg*)
  [0:25] alltoall_VPtest 0x55555585f93e ampi::delesend(int, int, void const*, int, int, int, int, CProxy_ampi, AmpiSendType, int)
  [0:26] alltoall_VPtest 0x555555823435 ampi::send(int, int, void const*, int, int, int, int, AmpiSendType, int)
  [0:27] alltoall_VPtest 0x555555826bc1 ampi::sendrecv(void const*, int, int, int, int, void*, int, int, int, int, int, MPI_Status*)
  [0:28] alltoall_VPtest 0x555555895d9b AMPI_Alltoall_long
  [0:29] alltoall_VPtest 0x555555807f27 AMPI_Main
  [0:30] alltoall_VPtest 0x5555558fb360 AMPI_Main_c
  [0:31] alltoall_VPtest 0x55555581b4fb AMPI_threadstart
  [0:32] alltoall_VPtest 0x555555808ef9 
  [0:33] alltoall_VPtest 0x555555a0ce7d CthStartThread
  [0:34] alltoall_VPtest 0x555555a0d30f make_fcontext
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI COMMUNICATOR 3 DUP FROM 0
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------

real	0m0.251s
user	0m0.250s
sys	0m0.077s
Makefile:38: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by evan-charmworks** [comment]
I believe the above problem is a bug in AMPI and could also explain strange failures seen in adcirc.

Unrelatedly, I've found a fix for the issue with alltoall and allgather, and pushed it accordingly.

**Pull request merged**